x/pocketcore/types/service.go

```diff
@@ -95,19 +122,60 @@ type Payload struct {
 	Headers map[string]string `json:"headers"` // http headers
 }
 
+func (p Payload) Bytes() []byte {
+	bz, err := json.Marshal(Payload{Data: p.Data, Method: p.Method, Path: p.Path})
+	if err != nil {
+		panic(fmt.Sprintf("an error occured converting the payload to bytes:\n%v", err))
+	}
+	return bz
+}
+
+func (p Payload) Hash() []byte {
+	return Hash(p.Bytes())
+}
+
+func (p Payload) HashString() string {
+	return hex.EncodeToString(p.Hash())
+}
+
 func (p Payload) Validate() sdk.Error {
 	if p.Data == "" && p.Path == "" {
 		return NewEmptyPayloadDataError(ModuleName)
 	}
 	return nil
 }
 
+type payload struct {
+	Data   string `json:"data"`
+	Method string `json:"method"`
+	Path   string `json:"path"`
+}
+
+func (p Payload) MarshalJSON() ([]byte, error) {
+	pay := payload{
+		Data:   p.Data,
+		Method: p.Method,
+		Path:   p.Path,
+	}
+	return json.Marshal(pay)
+}
+
+type RelayMeta struct {
+	BlockHeight int64 `json:"block_height"`
+}
+
+func (m RelayMeta) Validate(ctx sdk.Ctx) sdk.Error {
+	if ctx.BlockHeight()+5 < m.BlockHeight || ctx.BlockHeight()-5 > m.BlockHeight {
```

Comment:
Can we make a configuration for the `+5` `-5` from the get-go?

Reply:
Yes, but seems like a QOL separate issue kind of thing?

Reply:
https://github.com/pokt-network/pocket-core/issues/705